app/domain/operations/async_migrations/handlers/fa_application/create_application.rb

```diff
@@ -0,0 +1,152 @@
+          def find_application(application_id)
+            application = ::FinancialAssistance::Application.where(id: application_id).first
+            application ? Success(application) : Failure('Application not found')
+          end
+
+          def generate_new_draft_application(application)
+            copy_result = ::FinancialAssistance::Operations::Applications::Copy.new.call(
+              {
+                application_id: application.id,
+                origin: :migration,
+                generation_reason: :manual
+              }
+            )
+
+            copy_result.success? ? Success(copy_result.value!) : Failure("Failed to copy application: #{copy_result.failure}")
+          end
+
+          # @note This method:
+          #   * Generates individual market eligibilities and evidences for each applicant
+          #   * Migrates APTC/CSR eligibility data from old to new applicants
+          #   * Copies evidence records with appropriate type mapping
+          #   * Builds new evidences for each applicant
+          def generate_eligibilities(draft_application, application)
+            migrator = ::Migrations::DataModelMigrator.new
+            draft_application.applicants.each do |new_applicant|
+              old_applicant = application.applicants.select { |app| app.person_hbx_id == new_applicant.person_hbx_id }.first
+
+              person = Person.where(hbx_id: old_applicant.person_hbx_id).first
+              consumer_role = person.consumer_role
+              new_applicant.assign_attributes(
+                age_off_excluded: person.age_off_excluded,
+                contact_method: consumer_role.contact_method,
+                language_preference: consumer_role.language_preference
+              )
+
+              # Build individual_market_eligibility and its evidences
+              # social security number verification type ---> social security number evidence
+              # citizenship verification type ---> citizenship evidence
+              result = Operations::AsyncMigrations::Handlers::IndividualMarketEligibility::GenerateEvidences.new.call(applicant: new_applicant)
+              return Failure("Failed while generating applicant #{new_applicant.person_hbx_id} evidences #{result.failure}") if result.failure?
+
+              # Migrate existing aptc csr eligibility evidences
+              old_aptc_csr_eligibility = old_applicant.aptc_csr_eligibility
+              new_aptc_csr_eligibility = new_applicant.build_aptc_csr_eligibility
+              migrator.perform(old_aptc_csr_eligibility, new_aptc_csr_eligibility)
+              old_aptc_csr_eligibility.evidences.each do |old_evidence|
+                new_evidence = build_new_evidence(new_aptc_csr_eligibility, old_evidence)
+                migrator.perform(old_evidence, new_evidence)
+              end
+            end
+
+            Success(draft_application)
+          rescue StandardError => e
```

Comment:
Would it be helpful to have a logger file here?

Reply:
I will add loggers.

Reply:
I just checked we are already adding logger in the below subscriber. I think we can skip logger for this operation.

```
app/event_source/subscribers/batch_process_subscriber.rb

----subscribe(:on_process_migration_event)
```